src/ClientGenerator/ClientGenerator.cs

```diff
@@ -105,15 +106,23 @@ private bool CreateGrainClient(CodeGenOptions options)
                 Directory.CreateDirectory(outputFileDirectory);
             }
 
+            var services = new ServiceCollection();
+            services.AddSingleton<AssemblyProcessor>();
+            services.AddSingleton(sp => sp);
+            services.AddSingleton<IMessagingConfiguration>(new GlobalConfiguration());
+            services.AddSingleton<RoslynCodeGenerator>();
+            services.AddSingleton<SerializationManager>();
+            var serviceProvider = services.BuildServiceProvider();
+            var codeGenerator = serviceProvider.GetRequiredService<RoslynCodeGenerator>();
```

Comment:
There's some ugliness around this, by the way, since we don't link OrleansRuntime against the code generator and we have this `CodeGeneratorManager` type which tries to perform the linking at runtime